rest/oidc_api_test.go

```diff
@@ -1,139 +1,816 @@
 package rest
 
 import (
-	"io"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"regexp"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
+// A wantErrKind represents the specific error expected.
+type wantError struct {
+
+	// kind represents the specific kind of wantError expected.
+	kind wantErrKind
+
+	// code represents the expected status code from HTTP response.
+	code int
+
+	// message represents the expected message code from HTTP response.
+	message string
+}
+
+// A wantErrKind represents the specific kind of wantError.
+type wantErrKind int
+
+const (
+
+	// noCodeErr option forces the mock auth server to return the callback URL
+	// with no authorization code.
```

Comment:
It's not clear from these comments which requests the error is being forced for.  Switching 'forces the mock auth server...' to 'forces the _auth API ...' makes it clearer.